providers/oidc.go

```diff
@@ -123,6 +133,31 @@ func (p *OIDCProvider) enrichFromProfileURL(ctx context.Context, s *sessions.Ses
 	return nil
 }
 
+// enrichFromIntrospectURL enriches a session's claims and permissions via the JSON response of
+// an OIDC Introspection URL
+func (p *OIDCProvider) enrichFromIntrospectURL(ctx context.Context, s *sessions.SessionState) error {
+	clientSecret, err := p.GetClientSecret()
+	if err != nil {
+		return err
+	}
+	params := url.Values{}
```

Comment:
There's a lot going on here in the URL construction that should be split out into helper functions it looks like?

`providers/util.go` is an example where some of this might fit? The direct magic strings in the setup should likely use constants (either existing like `http.MethodPost` or ones you make if they don't exist).